src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateHelper.java

```diff
@@ -253,4 +269,97 @@ public static X500Name getX500Name(String commonName) {
 
         return nameBuilder.build();
     }
+
```

Comment:
Ported methods from Nucleus

Reply:
Why!?

a) You already have the Nucleus as a dependency, you can use any of its methods
b) There's a good reason why Nucleus abstracts these. These methods will not work when a HSM is used

Reply:
Don't port everything. Just take the certificate chain bit.

@MikeDombo The goal is to avoid bumping Nucleus dependency. I don't understand your HSM comment. The Nucleus wrapper is just calling the underlying providers, which is what should be happening here

Reply:
What is the current Nucleus dependency version? 

What functionality from the newer Nucleus version do you want?

Reply:
`Paths.get(privateKeyUri)` does not work when the key isn't a file. What key is going to be loaded by this method, do we know with certainty that it is going to be a file and not a pkcs11 URI?

Reply:
This code isn't correct. It needs to still use the SecurityService.

The new Nucleus API to get the cert chain is just a value. I want to avoid bumping Nucleus dependency for something we can easily include here. That method still uses other SecurityService primitives.

Right now CDA doesn't work with the latest public Nucleus. I want it to